src/Hazelcast.Net/Protocol/Codecs/SqlExecuteCodec.cs

```diff
@@ -0,0 +1,193 @@
+        private const int ResponseUpdateCountFieldOffset = Messaging.FrameFields.Offset.ResponseBackupAcks + BytesExtensions.SizeOfByte;
+        private const int ResponseInitialFrameSize = ResponseUpdateCountFieldOffset + BytesExtensions.SizeOfLong;
+
+#if SERVER_CODEC
+        public sealed class RequestParameters
+        {
+
+            /// <summary>
+            /// Query string.
+            ///</summary>
+            public string Sql { get; set; }
+
+            /// <summary>
+            /// Query parameters.
+            ///</summary>
+            public IList<IData> Parameters { get; set; }
+
+            /// <summary>
+            /// Timeout in milliseconds.
+            ///</summary>
+            public long TimeoutMillis { get; set; }
+
+            /// <summary>
+            /// Cursor buffer size.
+            ///</summary>
+            public int CursorBufferSize { get; set; }
+
+            /// <summary>
+            /// Schema name.
+            ///</summary>
+            public string Schema { get; set; }
+
+            /// <summary>
+            /// The expected result type. Possible values are:
+            ///   ANY(0)
+            ///   ROWS(1)
+            ///   UPDATE_COUNT(2)
+            ///</summary>
+            public byte ExpectedResultType { get; set; }
+
+            /// <summary>
+            /// Query ID.
+            ///</summary>
+            public Hazelcast.Sql.SqlQueryId QueryId { get; set; }
+        }
+#endif
+
+        public static ClientMessage EncodeRequest(string sql, ICollection<IData> parameters, long timeoutMillis, int cursorBufferSize, string schema, byte expectedResultType, Hazelcast.Sql.SqlQueryId queryId)
+        {
+            var clientMessage = new ClientMessage
+            {
+                IsRetryable = false,
+                OperationName = "Sql.Execute"
+            };
+            var initialFrame = new Frame(new byte[RequestInitialFrameSize], (FrameFlags) ClientMessageFlags.Unfragmented);
+            initialFrame.Bytes.WriteIntL(Messaging.FrameFields.Offset.MessageType, RequestMessageType);
+            initialFrame.Bytes.WriteIntL(Messaging.FrameFields.Offset.PartitionId, -1);
+            initialFrame.Bytes.WriteLongL(RequestTimeoutMillisFieldOffset, timeoutMillis);
+            initialFrame.Bytes.WriteIntL(RequestCursorBufferSizeFieldOffset, cursorBufferSize);
+            initialFrame.Bytes.WriteByteL(RequestExpectedResultTypeFieldOffset, expectedResultType);
+            clientMessage.Append(initialFrame);
+            StringCodec.Encode(clientMessage, sql);
+            ListMultiFrameCodec.EncodeContainsNullable(clientMessage, parameters, DataCodec.Encode);
+            CodecUtil.EncodeNullable(clientMessage, schema, StringCodec.Encode);
+            SqlQueryIdCodec.Encode(clientMessage, queryId);
+            return clientMessage;
+        }
+
+#if FALSE // FIXME [Olesii] discuss whether to remove these parts and how to update protocol
```

Comment:
Several members like `DecodeRequest` and `EncodeResponse` are not needed for these codecs and methods they are using are not implemented. Wanted to remove them, but that would make this different from code obtained from generate-codecs operation. At the moment, I don't see any option for C# generator to skip server-side methods for the codec. Should I update the protocol to support this or just remove this code and make it a bit different from auto-generate one?

Reply:
For other codecs, `DecodeRequest` and `EncodeResponse` are generated but within `#if SERVER_CODEC` blocks, so that they are *not* compiled in the main DLL but can be compiled in the testing DLL in case we want to simulate a server - which we do in some tests.

I think it is important that the SQL codecs are generated just like the others (via the generator), not custom.

What happens with these methods when the are generated, for SQL? Is the generated code not compilable? not runnable?

Reply:
It's became not compilable, because I didn't provide `SqlPage.Encode` method. I can add it but throw `NotSupportedException` inside, what do you think?

Reply:
Agreed on approach above.